### ISSUE TITLE
Build failure 'isnan' was not declared in this scope

### ISSUE THREAD

**Issue by jaf7** [xournalpp/xournalpp#167]
System information follows.  
```
[ 24%] Building CXX object src/CMakeFiles/xournalpp-core.dir/control/tools/VerticalToolHandler.cpp.o
/home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.cpp: In member function ‘UndoAction* EditSelectionContents::setFont(XojFont&)’:
/home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.cpp:174:16: error: ‘isnan’ was not declared in this scope
    if (isnan(x1))
                ^
/home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.cpp:174:16: note: suggested alternative:
In file included from /usr/include/c++/5.3.0/random:38:0,
                 from /usr/include/c++/5.3.0/bits/stl_algo.h:66,
                 from /usr/include/c++/5.3.0/algorithm:62,
                 from /usr/include/c++/5.3.0/ext/slist:47,
                 from /usr/include/boost/algorithm/string/std/slist_traits.hpp:16,
                 from /usr/include/boost/algorithm/string/std_containers_traits.hpp:23,
                 from /usr/include/boost/algorithm/string.hpp:18,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/util/StringUtils.h:16,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/Tool.h:14,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.h:16,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.cpp:1:
/usr/include/c++/5.3.0/cmath:641:5: note:   ‘std::isnan’
     isnan(_Tp __x)
     ^
/home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.cpp:202:15: error: ‘isnan’ was not declared in this scope
  if (!isnan(x1))
               ^
/home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.cpp:202:15: note: suggested alternative:
In file included from /usr/include/c++/5.3.0/random:38:0,
                 from /usr/include/c++/5.3.0/bits/stl_algo.h:66,
                 from /usr/include/c++/5.3.0/algorithm:62,
                 from /usr/include/c++/5.3.0/ext/slist:47,
                 from /usr/include/boost/algorithm/string/std/slist_traits.hpp:16,
                 from /usr/include/boost/algorithm/string/std_containers_traits.hpp:23,
                 from /usr/include/boost/algorithm/string.hpp:18,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/util/StringUtils.h:16,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/Tool.h:14,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.h:16,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.cpp:1:
/usr/include/c++/5.3.0/cmath:641:5: note:   ‘std::isnan’
     isnan(_Tp __x)
     ^
src/CMakeFiles/xournalpp-core.dir/build.make:518: recipe for target 'src/CMakeFiles/xournalpp-core.dir/control/tools/EditSelectionContents.cpp.o' failed
make[2]: *** [src/CMakeFiles/xournalpp-core.dir/control/tools/EditSelectionContents.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:348: recipe for target 'src/CMakeFiles/xournalpp-core.dir/all' failed
make[1]: *** [src/CMakeFiles/xournalpp-core.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```
$ `uname -a`:
Linux _nodename_ 4.4.1-2-surfacepro3 \# 1 SMP PREEMPT Fri Mar 18 15:55:31 MDT 2016 x86_64 GNU/Linux
`gcc -c -Q -march=haswell --help=target | grep enabled`:
  -m64                        		[enabled]
  -m80387                     		[enabled]
  -m96bit-long-double         	[enabled]
  -malign-stringops           	[enabled]
  -mfancy-math-387            	[enabled]
  -mfentry                    		[enabled]
  -mfp-ret-in-387             		[enabled]
  -mglibc                     		[enabled]
  -mhard-float                		[enabled]
  -mieee-fp                   		[enabled]
  -mlong-double-80            	[enabled]
  -mno-sse4                   		[enabled]
  -mpush-args                 		[enabled]
  -mred-zone                  		[enabled]
  -mstackrealign              		[enabled]
  -mtls-direct-seg-refs       	[enabled]
`$ cat /proc/cpuinfo | grep -m 1 flags`:
flags	: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm ida arat epb pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid xsaveopt

**Comment by ziekke** [comment]
I am experiencing the same identical error

Linux nodename 4.3.3-2-surfacepro3 #1 SMP PREEMPT Tue Feb 16 14:41:45 EST 2016 x86_64 GNU/Linux

**Comment by andreasb242** [comment]
Can you try replace the `isnan` with `::isnan`?

Maybe there is a problem with the Namespace, then a two colons means using no namespace.

std::isnan was added with C++11, so it's a new feature.

**Comment by ziekke** [comment]
I just get: 

error: ‘::isnan’ has not been declared
    if (::isnan(x1))
[...]

**Comment by MarPiRK** [comment]
What version of gcc you have? Does it support C++11?
Also try to add:
```C++
#include <cmath>
```
at the begging of the file.

**Comment by ziekke** [comment]
```
gcc --version
gcc (GCC) 5.3.0
```
 whether it supports C++11 I'm not sure. I'll add the include and see

**Comment by ziekke** [comment]
Okay I got it.
I made the following changes:

replace
```c++
#include <math.h>
```
with
```c++
#include <cmath>
```

and I replaced all instances of 
```c++
isnan
```
with
```c++
std:isnan
```

My apologies if that's what andreas intended, but the build seems to have progressed past the point where it was failing.

**Comment by jaf7** [comment]
Ok, thanks!  Successful build here too, making almost exactly same changes as @ziekke.  (Double :: instead of single :) Repaced:
`#include <math.h>`
with:
`#include <cmath>`
and all instances of 
`isnan`
with:
`std::isnan`
[This](http://stackoverflow.com/questions/18128899/is-isnan-in-the-std-namespace-more-in-general-when-is-std-necessary-optio) looks like it might directly address this issue (Stack Overflow, isnan in the std:: namespace).
  And [this is referenced](http://en.cppreference.com/w/cpp/numeric/math/isnan) in the above SO answers (relevant?):
> Another way to test if a floating-point value is NaN is to compare it with itself: 
`bool is_nan(double x) { return x != x; }`

With fresh sources, just like @ziekke I get same error with `::isnan`
Compiler does suggest `std::isnan` from cmath (?):
`/home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.cpp:174:8: note: suggested alternative:
In file included from /usr/include/c++/5.3.0/random:38:0,
                 from /usr/include/c++/5.3.0/bits/stl_algo.h:66,
                 from /usr/include/c++/5.3.0/algorithm:62,
                 from /usr/include/c++/5.3.0/ext/slist:47,
                 from /usr/include/boost/algorithm/string/std/slist_traits.hpp:16,
                 from /usr/include/boost/algorithm/string/std_containers_traits.hpp:23,
                 from /usr/include/boost/algorithm/string.hpp:18,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/util/StringUtils.h:16,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/Tool.h:14,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.h:16,
                 from /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/src/control/tools/EditSelectionContents.cpp:1:
/usr/include/c++/5.3.0/cmath:641:5: note:   ‘std::isnan’`

So, successful build on Arch using `makepkg -so`, editing, then `makepkg -er`:
`Configuration:
        Compiler:                   /usr/bin/c++
        Overlay Scrollbar enabled:  OFF
        Mathtex enabled:            ON
        CppUnit enabled:            OFF

Unstable features:
        Layers sidebar:             OFF

Static libraries:
        Poppler:                    OFF
        Boost:                      OFF
        Glibmm:                     OFF

-- Configuring done
-- Generating done
-- Build files have been written to: /home/mien/builds/build_tmp/xournalpp-git/src/xournalpp/build
==> Starting pkgver()...
==> Updated version: xournalpp-git 1.0.0.317.g283273c-1
==> Sources are ready.

...

==> Finished making: xournalpp-git 1.0.0.317.g283273c-1 (Mon Mar 21 19:57:15 MDT 2016)`

**Comment by zParquet** [comment]
> Okay I got it. I made the following changes:
> 
> replace
> 
> ```c++
> #include <math.h>
> ```
> 
> with
> 
> ```c++
> #include <cmath>
> ```
> 
> and I replaced all instances of
> 
> ```c++
> isnan
> ```
> 
> with
> 
> ```c++
> std:isnan
> ```
> 
> My apologies if that's what andreas intended, but the build seems to have progressed past the point where it was failing.

It works! thanks a lot !
